src/Aspire.Hosting/ApplicationModel/ReplicaInstancesAnnotation.cs

```diff
@@ -10,5 +10,5 @@ namespace Aspire.Hosting.ApplicationModel;
 /// </summary>
 internal class ReplicaInstancesAnnotation : IResourceAnnotation
 {
-    public ConcurrentDictionary<string, string> Instances { get; } = [];
+    public ConcurrentDictionary<string, int> Instances { get; } = [];
```

Comment:
int = replica index.

I was using this at one point, but it can be reverted. I left it in in case there is a reason we want to know replica instance from the `IResource` layer.